flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java

```diff
@@ -111,9 +112,10 @@ public void testConsumeWithoutStartSnapshotId() throws Exception {
       TestSourceContext sourceContext = new TestSourceContext(latch);
       runSourceFunctionInTask(sourceContext, function);
 
-      Assert.assertTrue(
-          "Should have expected elements.", latch.await(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS));
-      Thread.sleep(1000L);
+      Awaitility.await()
+        .atMost(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS)
+        .pollInterval(100, TimeUnit.MILLISECONDS)
+        .until(() -> latch.getCount() == 0);
```

Comment:
I don't think this is a good idea.  There are high changes to become flaky tests. 
At bad day, your latch count became 0 and threads are still doing remaining processing. but your worker thread will start compiling next lines. 
I think let `latch.await` be there . Just replace Thread.sleep with `Awaitility` .